File: tests/fixtures/java-spark/src/main/java/com/example/EventsJob.java
package com.example;

import org.apache.spark.sql.Dataset;
import org.apache.spark.sql.Row;
import org.apache.spark.sql.SparkSession;

public class EventsJob {
    public static Dataset<Row> loadEvents(SparkSession spark, String path) {
        return spark.read().parquet(path);
    }

    public static void saveEvents(Dataset<Row> df, String path) {
        df.write().mode("overwrite").parquet(path);
    }

    public static Dataset<Row> countByStatus(Dataset<Row> df) {
        return df.groupBy("status").count();
    }
}

File: tests/fixtures/scala-spark/src/main/scala/com/example/EventsJob.scala
package com.example

import org.apache.spark.sql.{DataFrame, SparkSession}

object EventsJob {
  def loadEvents(spark: SparkSession, path: String): DataFrame =
    spark.read.parquet(path)

  def saveEvents(df: DataFrame, path: String): Unit =
    df.write.mode("overwrite").parquet(path)

  def countByStatus(df: DataFrame): DataFrame =
    df.groupBy("status").count()
}
